src/Tools/SchemaTool.php

```diff
@@ -337,7 +337,7 @@ public function getSchemaFromMetadata(array $classes): Schema
 
             if (isset($class->table['uniqueConstraints'])) {
                 foreach ($class->table['uniqueConstraints'] as $indexName => $indexData) {
-                    $uniqIndex = new Index($indexName, $this->getIndexColumns($class, $indexData), true, false, [], $indexData['options'] ?? []);
+                    $uniqIndex = new Index(is_numeric($indexName) ? null : $indexName, $this->getIndexColumns($class, $indexData), true, false, [], $indexData['options'] ?? []);
```

Comment:
Passing `null` here would cause errors with DBAL 3.

The `Index` object will be destroyed a few lines later anyway, so a string cast might just be enough.